gds_write/build_defs.bzl

```diff
@@ -0,0 +1,127 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""These build rules run automated GDS write on an implemented design"""
+
+load("//place_and_route:open_road.bzl", "OpenRoadInfo")
+load("//synthesis:build_defs.bzl", "SynthesisInfo")
+
+def _gds_write_impl(ctx):
+    final_gds = ctx.actions.declare_file("{}.gds".format(ctx.attr.name))
+    klayout_final_lyt = ctx.actions.declare_file("{}_klayout.lyt".format(ctx.attr.name))
+    klayout_final_lef = ctx.actions.declare_file("{}_klayout.lef".format(ctx.attr.name))
+    # Declare a list of additional LEF files to be copied into bazel build directory
+    final_lefs = []
+    for lef in ctx.files.additional_lef:
+        final_lefs.append(ctx.actions.declare_file("{}_{}.lef".format(ctx.attr.name, lef.basename.split('.', 1)[0])))
+
+    # Copy additional LEF files to bazel build directory
```

Comment:
curious why we need to copy the lef file? can't we just use them from their current location?

Reply:
Managed to simplify the rule and remove copying files